api-gateway/Jenkinsfile

```diff
@@ -0,0 +1,111 @@
+                    COMMITTER_NAME = commitInfo[0]?.trim() ?: 'Unknown'
+
+                    def userCause = currentBuild.rawBuild.getCause(hudson.model.Cause$UserIdCause)
+                    if (userCause) {
+                        TRIGGERED_BY = "Triggered manually by: ${userCause.getUserName()}"
+                    } else {
+                        TRIGGERED_BY = "Triggered by: ${COMMITTER_NAME}"
+                    }
+                    echo TRIGGERED_BY
+                }
+            }
+        }
+
+        stage('Checkout') {
+            steps {
+                script {
+                    checkout scm: [
+                        $class: 'GitSCM',
+                        branches: [[name: "*/${params.BRANCH_NAME}"]],
+                        doGenerateSubmoduleConfigurations: false,
+                        extensions: [],
+                        userRemoteConfigs: [[url: 'https://github.com/imzainazm/microservices-chat-demo.git']]
+                    ]
+                    currentBuild.description = TRIGGERED_BY
+                    echo TRIGGERED_BY
+                }
+            }
+        }
+
+        stage('Build Docker Image') {
+            steps {
+                script {
+                    def shortCommitHash = sh(returnStdout: true, script: 'git rev-parse --short HEAD').trim()
+                    dockerBuild("imzainazm/api-gateway", shortCommitHash, "./api-gateway")
+                }
+            }
+        }
+
+        stage('Push Docker Image to Docker Hub') {
+            steps {
+                script {
+                    def shortCommitHash = sh(returnStdout: true, script: 'git rev-parse --short HEAD').trim()
+                    dockerPush("imzainazm/api-gateway", shortCommitHash)
+                }
+            }
+        }
+    }
+
+    post {
+        success {
+            script {
+                sendSlackNotification(true)
+                cleanupImages()
+            }
+        }
+        failure {
+            script {
+                sendSlackNotification(false)
+                cleanupImages()
+            }
+        }
+    }
+}
+
+def dockerBuild(imageName, tag, dockerfilePath) {
+    sh "docker build -t ${imageName}:latest -t ${imageName}:${tag} ${dockerfilePath}"
+}
+
+def dockerPush(imageName, tag) {
+    docker.withRegistry('https://index.docker.io/v1/', DOCKER_CREDENTIALS_ID) {
+        docker.image("${imageName}:latest").push()
+        docker.image("${imageName}:${tag}").push()
+    }
+}
+
+def sendSlackNotification(isSuccess) {
+    def pipelineStatus = isSuccess ? "Succeeded" : "Failed"
+    def environmentName = env.JOB_NAME.split('/')[0] ?: 'Unknown'
+
+    slackSend(
+        botUser: true,
+        channel: SLACK_CHANNEL,
+        color: isSuccess ? '#00ff00' : '#ff0000',
+        message: "Pipeline ${pipelineStatus}\n${TRIGGERED_BY}\nEnvironment: ${environmentName}",
```

Comment:
Add some deployment info please. Branch name, commit message etc. Whatever do you think a teammate of manager would require to understand whats being deployed.

Reply:
Added branch name and pipeline build number as well in the slack message.